iqm5q.py

```diff
@@ -17,65 +20,74 @@
-        )
-
-    hdawg = []
-    for i in range(5):
-        hdawg.append({"rf_signal": f"q{i}/flux_line", "ports": f"SIGOUTS/{i}"})
-    for c, i in zip(itertools.chain(range(0, 2), range(3, 4)), range(5, 8)):
-        hdawg.append({"rf_signal": f"qc{c}/flux_line", "ports": f"SIGOUTS/{i}"})
-
-    hdawg2 = [{"rf_signal": "qc4/flux_line", "ports": f"SIGOUTS/0"}]
-
-    pqsc = [
-        "internal_clock_signal",
-        {"to": "device_hdawg2", "port": "ZSYNCS/1"},
-        {"to": "device_hdawg", "port": "ZSYNCS/0"},
-        {"to": "device_shfqc", "port": "ZSYNCS/2"},
-    ]
+    device_setup = DeviceSetup("EL_ZURO")
+    # Dataserver
+    device_setup.add_dataserver(host="localhost", port=8004)
+    # Instruments
+    device_setup.add_instruments(
+        HDAWG("device_hdawg", address="DEV8660"),
+        HDAWG("device_hdawg2", address="DEV8673"),
+        PQSC("device_pqsc", address="DEV10055", reference_clock_source="internal"),
+        SHFQC("device_shfqc", address="DEV12146"),
+    )
+    device_setup.add_connections(
+        "device_shfqc",
+        *[
+            create_connection(
+                to_signal=f"q{i}/drive_line", ports=[f"SGCHANNELS/{i}/OUTPUT"]
+            )
+            for i in range(N_QUBITS)
+        ],
+        *[
+            create_connection(
+                to_signal=f"q{i}/measure_line", ports=["QACHANNELS/0/OUTPUT"]
+            )
+            for i in range(N_QUBITS)
+        ],
+        *[
+            create_connection(
+                to_signal=f"q{i}/acquire_line", ports=["QACHANNELS/0/INPUT"]
+            )
+            for i in range(N_QUBITS)
+        ],
+    )
+    device_setup.add_connections(
+        "device_hdawg",
+        *[
+            create_connection(to_signal=f"q{i}/flux_line", ports=f"SIGOUTS/{i}")
+            for i in range(N_QUBITS)
+        ],
+        *[
+            create_connection(to_signal=f"qc{c}/flux_line", ports=f"SIGOUTS/{i}")
+            for c, i in zip(itertools.chain(range(0, 2), range(3, 4)), range(5, 8))
+        ],
+    )
 
-    connections = {
-        "device_shfqc": shfqc,
-        "device_hdawg": hdawg,
-        "device_hdawg2": hdawg2,
-        "device_pqsc": pqsc,
-    }
+    device_setup.add_connections(
+        "device_hdawg2",
+        create_connection(to_signal="qc4/flux_line", ports=["SIGOUTS/0"]),
+    )
 
-    descriptor = {
-        "instruments": instruments,
-        "connections": connections,
-    }
+    device_setup.add_connections(
+        "device_pqsc",
+        create_connection(to_instrument="device_hdawg2", ports="ZSYNCS/1"),
+        create_connection(to_instrument="device_hdawg", ports="ZSYNCS/0"),
+        create_connection(to_instrument="device_shfqc", ports="ZSYNCS/2"),
+    )
 
     controller = Zurich(
         "EL_ZURO",
-        descriptor,
+        device_setup=device_setup,
         use_emulation=False,
         time_of_flight=75,
         smearing=50,
```

Comment:
@GabrielePalazzo I think the following interface would be neater. What do you think? I also left a comment in the parallel PR: https://github.com/qiboteam/qibolab/pull/664

``` py
controller = Zurich.from_device_setup(device_setup, "EL_ZURO", use_emulation=False, time_of_flight=75, smearing=50)
```